src/momento/requests/vector_index/filters.py

```diff
@@ -0,0 +1,300 @@
+            return vectorindex_pb._GreaterOrEqualExpression(field=self.field, float_value=self.value)
+        else:
+            raise InvalidArgumentException(
+                f"Invalid type for value: {type(self.value)} in greater than or equal expression. Must be one of int, float.",
+                Service.INDEX,
+            )
+
+
+@dataclass
+class LessThan(FilterExpression):
+    """Represents a less than expression between a field and a value."""
+
+    field: str
+    """The field to compare."""
+    value: int | float
+    """The value to test less than with."""
+
+    def to_filter_expression_proto(self) -> vectorindex_pb._FilterExpression:
+        return vectorindex_pb._FilterExpression(less_than_expression=self.to_proto())
+
+    def to_proto(self) -> vectorindex_pb._LessThanExpression:
+        if type(self.value) is int:
+            return vectorindex_pb._LessThanExpression(field=self.field, integer_value=self.value)
+        elif type(self.value) is float:
+            return vectorindex_pb._LessThanExpression(field=self.field, float_value=self.value)
+        else:
+            raise InvalidArgumentException(
+                f"Invalid type for value: {type(self.value)} in less than expression. Must be one of int, float.",
+                Service.INDEX,
+            )
+
+
+@dataclass
+class LessThanOrEqual(FilterExpression):
+    """Represents a less than or equal expression between a field and a value."""
+
+    field: str
+    """The field to compare."""
+    value: int | float
+    """The value to test less than or equal with."""
+
+    def to_filter_expression_proto(self) -> vectorindex_pb._FilterExpression:
+        return vectorindex_pb._FilterExpression(less_or_equal_expression=self.to_proto())
+
+    def to_proto(self) -> vectorindex_pb._LessOrEqualExpression:
+        if type(self.value) is int:
+            return vectorindex_pb._LessOrEqualExpression(field=self.field, integer_value=self.value)
+        elif type(self.value) is float:
+            return vectorindex_pb._LessOrEqualExpression(field=self.field, float_value=self.value)
+        else:
+            raise InvalidArgumentException(
+                f"Invalid type for value: {type(self.value)} in less than or equal expression. Must be one of int, float.",
+                Service.INDEX,
+            )
+
+
+@dataclass
+class ListContains(FilterExpression):
+    """Represents a list contains expression between a field and a value."""
+
+    field: str
+    """The list field to test."""
+    value: str
+    """The value to test list contains with."""
+
+    def to_filter_expression_proto(self) -> vectorindex_pb._FilterExpression:
+        return vectorindex_pb._FilterExpression(list_contains_expression=self.to_proto())
+
+    def to_proto(self) -> vectorindex_pb._ListContainsExpression:
+        # todo should make oneof defensively
+        return vectorindex_pb._ListContainsExpression(field=self.field, value=self.value)
+
+
+@dataclass
+class Field:
```

Comment:
Alternate names if this feels too non-specific:
- `FilterField`
- `ExpressionField`

Reply:
i think the name is okay, I like the way it reads in your examples. i wonder though if it might be worth putting it in a separate file just to make it more obvious how distinct it is from all of the code above it in this file though?

just a thought, I don't feel strongly.

Reply:
Sounds good 👍

Reply:
Done in [545bdb6](https://github.com/momentohq/client-sdk-python/pull/425/commits/545bdb6c493e7dd49e8d2d58b718071513ad2ba0)